code/controllers/subsystems/records.dm

```diff
@@ -0,0 +1,137 @@
+/var/datum/controller/subsystem/records/SSrecords
+
+/datum/controller/subsystem/records
+    name = "Records"
+    flags = SS_NO_FIRE
+
+    var/list/records
+
+/datum/controller/subsystem/records/New()
+    LAZYINITLIST(records)
+    NEW_SS_GLOBAL(SSrecords)
+
+/datum/controller/subsystem/records/proc/generate_record(var/mob/living/carbon/human/H)
+    if(H.mind && !player_is_antag(H.mind, only_offstation_roles = 1) && SSjobs.ShouldCreateRecords(H.mind.assigned_role))
+        var/assignment = GetAssignment(H)
+        var/id = generate_record_id()
+        var/list/r = create_empty_record(H, id)
```

Comment:
I wonder, is there a point in maintaining records as lists? It leaves the entire system vulnerable to key errors and related shit. Datumizing them would also make the code a bit cleaner.